### PR TITLE
chapter 9

### PR DESCRIPTION
**Pull request by maihuysl1999**
Advance Login 
Kiến thức cần nắm: https://docs.google.com/document/d/1L9dlWmhGjGeCMPiZD5of0TF-LqGJJK9PEh6HX9fVXcY/edit
Screenshot from 2021-08-03 21-38-27

### CODE REVIEW

**Comment by maihuysl1999** [comment]
ready

File: app/helpers/sessions_helper.rb [hunk 1]
```diff
@@ -4,14 +4,35 @@ def log_in user
   end
 
   def current_user
-    @current_user ||= User.find_by id: session[:user_id]
+    if (user_id = session[:user_id])
```
Comment (ngocvt-0484) [inline app/helpers/sessions_helper.rb]:
```suggestion
    if user_id = session[:user_id]
```

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -4,14 +4,35 @@ def log_in user
...
+      @current_user ||= User.find_by id: user_id
+    elsif (user_id = cookies.signed[:user_id])
```
Comment (ngocvt-0484) [inline app/helpers/sessions_helper.rb]:
```suggestion
    elsif user_id = cookies.signed[:user_id]
```

File: app/helpers/sessions_helper.rb [hunk 3]
```diff
@@ -4,14 +4,35 @@ def log_in user
...
+      user = User.find_by id: user_id
+      if user&.authenticated?(cookies[:remember_token])
```
Comment (ngocvt-0484) [inline app/helpers/sessions_helper.rb]:
```suggestion
      if user&.authenticated? cookies[:remember_token]
```
chỗ nào bỏ dc ngoặc thì bỏ e nhé

File: app/helpers/sessions_helper.rb [hunk 4]
```diff
@@ -4,14 +4,35 @@ def log_in user
...
 
+  def remember user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
+  def forget user
+    user.forget
+    cookies.delete(:user_id)
```
Comment (ngocvt-0484) [inline app/helpers/sessions_helper.rb]: same

**Comment by maihuysl1999** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**